File: 0700-0799/0700. Search in a Binary Search Tree/0700. Search in a Binary Search Tree.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


from typing import Optional


class Solution:
    def searchBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        if (not root) or root.val == val:
            return root
        else:
            return self.searchBST(root.left, val) if val < root.val else self.searchBST(root.right, val)

File: 0700-0799/0700. Search in a Binary Search Tree/0700. Search in a Binary Search Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
const searchBST = function (root, val) {
    if (root === null || root.val === val) return root
    else return val < root.val ? searchBST(root.left, val) : searchBST(root.right, val)
};

const searchBST2 = function (root, val) {
    while (root !== null && root.val !== val) root = val < root.val ? root.left : root.right;
    return root;
};